File: Python/4-ISP/ISP-Violacao/CadastroProduto.py
from ICadastro import ICadastro

class CadastroProduto(ICadastro):
    
    def validarDados(self):
        print('Validar dados')
    
    def salvarBanco(self):
         print('Salvar banco')
    
    def enviarEmail(self):
        # Produto não tem e-mail, o que eu faço agora???
        raise NotImplementedError('Esse método não serve pra nada')

File: PHP/4-ISP/ISP-Violacao/CadastroProduto.php
<?php
require_once 'ICadastro.php';
    
class CadastroProduto implements ICadastro
{
    public function validarDados()
    {
        // Validar valor
        echo "\n validar dados";
    }

    public function salvarBanco()
    {
        // Insert tabela Produto
        echo "\n salvar banco";
    }

    public function enviarEmail()
    {
        // Produto não tem e-mail, o que eu faço agora???
        echo "\n";
        throw new NotImplementedException("Esse metodo não serve pra nada");
    }
}
